tests/conftest.py

```diff
@@ -4618,3 +4619,61 @@ def finalizer():
 
     request.addfinalizer(finalizer)
     return factory
+
+
+@pytest.fixture(scope="session", autouse=True)
+def patch_consumer_toolbox_with_secret():
+    """
+    Patch the rook-ceph-tools deployment with ceph.admin key. Applicable for MS platform only to enable rook-ceph-tools
+    to run ceph commands until we have the fix for rook-ceph-tools in consumer cluster
+
+    """
+    current_cluster = config.cluster_ctx
+
+    # Get the secret from provider if MS multicluster run
+    if (
```

Comment:
Can we add another if block to get the key from env_data if present? @jilju ?

Reply:
Added check for the environment variable CEPHADMINKEY. The key can be added as a env variable or in ocs-ci/data/auth.yaml file.

Use the template given below to add the key in auth.yaml
```
external:
  ceph_admin_key: "key"
```

Reply:
If the key is not present in env (CEPHADMINKEY) or in auth.yaml , the fixture will need to access the provider cluster. In this case provider cluster details should be passed in the run-ci command.